File: Answer/3689.最大子数组总值-i.py
#
# @lc app=leetcode.cn id=3689 lang=python3
#
# [3689] 最大子数组总值 I
#

# @lc code=start
from typing import List


class Solution:
    def maxTotalValue(self, nums: List[int], k: int) -> int:
        return (max(nums) - min(nums)) * k
# @lc code=end

File: Answer/3689.最大子数组总值-i.cs
/*
 * @lc app=leetcode.cn id=3689 lang=csharp
 *
 * [3689] 最大子数组总值 I
 */

// @lc code=start
public class Solution
{
    public long MaxTotalValue(int[] nums, int k)
    {
        long min = nums[0];
        long max = nums[0];
        for (var i = 1; i < nums.Length; i++)
        {
            long v = nums[i];
            if (v < min)
            {
                min = v;
            }
            if (v > max)
            {
                max = v;
            }
        }
        return (max - min) * k;
    }
}
// @lc code=end
